### PR TITLE
[Aae-16885] h2 definitions fix

### PR DESCRIPTION
**Pull request by wojciech-piotrowiak**

### CODE REVIEW

File: activiti-core/activiti-engine/src/test/resources/META-INF/persistence.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
     <exclude-unlisted-classes>true</exclude-unlisted-classes>
     <properties>
         <property name="jakarta.persistence.jdbc.driver" value="org.h2.Driver" />
-        <property name="jakarta.persistence.jdbc.url" value="jdbc:h2:mem:activiti-jpa;DB_CLOSE_DELAY=1000" />
+        <property name="jakarta.persistence.jdbc.url" value="jdbc:h2:mem:activiti-jpa;DB_CLOSE_DELAY=1000;NON_KEYWORDS=VALUE" />
```
Comment (erdemedeiros) [inline activiti-core/activiti-engine/src/test/resources/META-INF/persistence.xml:37]: Why is this needed? Which use cases is it covering? The [documentation](https://www.h2database.com/html/commands.html#set_non_keywords) suggests to use quoted identifiers if possible.

**Review by erdemedeiros — COMMENTED** [review]
Hi, I have one question concerning `NON_KEYWORDS`. LGTM, otherwise.

File: activiti-core/activiti-engine/src/test/resources/META-INF/persistence.xml (hunk 1 above)
Comment (wojciech-piotrowiak) [reply activiti-core/activiti-engine/src/test/resources/META-INF/persistence.xml:37]: Tried to quote column 'value' in FieldAccessJPAEntity but it does not help

Comment (erdemedeiros) [reply activiti-core/activiti-engine/src/test/resources/META-INF/persistence.xml:37]: Ok, thank you for trying it. Let's keep with `NON_KEYWORDS` in this case.

**Review by erdemedeiros — APPROVED** [review]
LGTM!
